p_sets/p_set_1.rb

```diff
@@ -0,0 +1,91 @@
+# http://cdn.cs50.net/2016/x/psets/1/pset1/pset1.html
+# SMART WATER
+BOTTLE_VOLUME = 16
+WATER_FLOW_RATE = 192
+
+def bottle_reporter
+  puts "That's #{bottle_calculator ask_minutes} bottles of water"
```

Comment:
Like C has `sprintf`, so does Ruby, aliased as `format`. While it's not *always* the best option, it can help split out the format of a string from the data that's being embedded into it:
```
format("That's %i bottles of water", bottle_calculator(ask_minutes)) # 'i' specifies the argument is an integer
```

If you have data in a Hash already then it's also possible to name the parameters rather than rely on them being passed in the same order as the placeholders:

```
format("That's %<bottles>i bottles of water",
       bottles: bottle_calculator(ask_minutes))
```

but to be clear, I'm only mentioning it as a way to simplify more complicated `format` calls, for example when there are multiple arguments. I don't recommend that as a first step.